### PR TITLE
Add split violin partially

### PR DESCRIPTION
**Pull request by ahuang11**
https://github.com/holoviz/holoviews/issues/4096
Not ready yet. I don't know how to add a legend and change the colors. Not sure if this implementation is the best approach. Also, not sure how to make the widths as proportionate as seaborn. Feel free to edit/make commits directly. Also didn't handle for matplotlib/plotly.

```
import numpy as np
import holoviews as hv
hv.extension('bokeh')

import seaborn as sns
sns.set(style="whitegrid")
tips = sns.load_dataset("tips")

ax = sns.violinplot(x="day", y="total_bill", hue="sex",
                    data=tips, palette="Set2", split=True,
                    scale="count")

import pandas as pd
tips = pd.concat([
    tips.loc[tips['day'] == 'Thur'],
    tips.loc[tips['day'] == 'Fri'],
    tips.loc[tips['day'] == 'Sat'],
    tips.loc[tips['day'] == 'Sun']
])

hv.Violin(tips, 'day', ['total_bill', 'sex']).opts(ylim=(-10, 60), width=600)
```

image

### CODE REVIEW

File: holoviews/plotting/bokeh/stats.py [hunk 1]
```diff
@@ -342,18 +342,52 @@ class ViolinPlot(BoxWhiskerPlot):
     selection_display = BokehOverlaySelectionDisplay(color_prop='violin_fill_color')
 
     def _kde_data(self, el, key, **kwargs):
-        vdim = el.vdims[0]
-        values = el.dimension_values(vdim)
+        vdims = el.vdims
+        vdim = vdims[0]
```
Comment (philippjfr) [inline holoviews/plotting/bokeh/stats.py]: -1 on using vdims for this, HoloViews is almost entirely built around the idea that the data is organized in a tidy format. It'd have to be an additional kdim which can optionally be mapped using a plot option.

**Comment by philippjfr** [comment]
Thanks for having a go at this, sadly as I said above this design doesn't really fit in with the rest of HoloViews.

**Comment by ahuang11** [comment]
No worries. I was basing it off the hv.Area(x, y, y1) idea `hv.Area((X, Y, Y2), vdims=['y', 'y2']) * hv.Area((X, Y, Y3), vdims=['y', 'y3'])` and think I tried kdims first, but it created groups like:
image

**Comment by ahuang11** [comment]
Okay, I swapped vdims for kdims. Still not sure how to add coloring/legend.
image

**Comment by philippjfr** [comment]
Thanks, I'll take a look at coloring and legend. Instead of split=True/False I think it should accept a dim mapping.

**Comment by philippjfr** [comment]
Initial attempt, still needs some cleanup:

Screen Shot 2020-01-13 at 5 43 52 PM

**Comment by philippjfr** [comment]
Added a gallery example:

bokeh_plot - 2020-01-13T184317 102

**Comment by ahuang11** [comment]
Really excited for this!

**Comment by ccneko** [comment]
Thanks for the really nice functionality. By the way, any chance of Plotly support? (Looking at Panel to use both Bokeh & Plotly meanwhile)

**Comment by ahuang11** [comment]
I don't think so, but maybe you can open a new issue for split violin plots in plotly.

**Comment by ccneko** [comment]
@ahuang11 Thanks for confirming. I will directly use the Plotly split violin plots first for the current usage.

**Pull request merged**